hw-model/src/lib.rs

```diff
@@ -585,6 +585,15 @@ pub trait HwModel: SocManager {
             const MAX_WAIT_CYCLES: u32 = 20_000_000;
             let mut cycles = 0;
             while !self.ready_for_fw() {
+                // If GENERATE_IDEVID_CSR was set then we need to clear cptra_dbg_manuf_service_reg
+                // once the CSR is ready to continue making progress.
+                //
+                // Generally the CSR should be read from the mailbox at this point, but to
+                // accommodate test cases that ignore the CSR mailbox, we will ignore it here.
+                if self.soc_ifc().cptra_flow_status().read().idevid_csr_ready() {
+                    self.soc_ifc().cptra_dbg_manuf_service_reg().write(|_| 0);
+                }
```

Comment:
For tests which set the bit, I wonder if they should be responsible for clearing it (even if they're not going to read the CSR). I think it makes the expected behavior when reading the test a bit less magic.

Reply:
We chatted offline and decided to keep this. 

It's unfortunate that this coupled logic is buried but the other options involve a log of duplicate logic or large refactor.